homeassistant/components/openweathermap/sensor.py

```diff
@@ -309,11 +318,39 @@ def __init__(
         """Initialize the sensor."""
         super().__init__(name, unique_id, description, weather_coordinator)
         self._weather_coordinator = weather_coordinator
+        self._update_attr()
 
-    @property
-    def native_value(self) -> StateType:
-        """Return the state of the device."""
-        return self._weather_coordinator.data.get(self.entity_description.key, None)
+    def _update_attr(self) -> None:
+        """Update _attr."""
+        data = self._weather_coordinator.data.get(self.entity_description.key, None)
+
+        if self.entity_description.key == ATTR_API_NATIONAL_WEATHER_ALERTS:
+            if data is None:
+                self._attr_native_value = 0
+                self._attr_extra_state_attributes = {"alerts": None}
+                return
+            self._attr_native_value = len(data)
+
+            variable_safe = []
+            for alert in data:
+                alert_data = {
+                    "sender": alert.sender,
+                    "title": alert.title,
+                    "start": str(alert.start_time()),
+                    "end": str(alert.end_time()),
+                    "description": alert.description,
+                    "tags": alert.tags,
+                }
+                variable_safe.append(alert_data)
+
+            self._attr_extra_state_attributes = {"alerts": variable_safe}
```

Comment:
Storing all the alerts with their data in the attributes will blow up the state machine.
We can leave the number of alerts as sensor value, but we should get the alert data via a service call, similar to the weather forecast.

Reply:
Good point. Could we fire an event when a new weather alert is issued? Some automations depend on knowing when an alert is issued ASAP.

Reply:
You could have another sensor with a timestamp of when latest alert came in or something like that which they can trigger on. I personally think events should be avoided as they are more difficult than a normal entity to use for automations etc.

Reply:
That sounds more like an event entity

Reply:
I'm just confused on how to detect when a new alert comes in. Possibly:

- Monitor the alert data in a variable in an event entity
- When an update comes in, compare the new data versus the old data
- If it's different, fire the event entity with what's new?

Reply:
Never mind, just looked at the update interval and that's way too long in a situation where seconds could count. Putting it in a service call would work

Reply:
Hmm, the way I'm thinking of it, can there be more alerts at the same time? Then a calendar would be ideal as it can hold more alerts at the same time.

If there can be only one alert at a time, an event entity would be the best, as you can create an event for weather alert announced, active and inactive.

Reply:
Oh didn't see your reaction, that would work as well :)